mula/scheduler/connectors/errors.py

```diff
@@ -12,7 +13,11 @@ def exception_handler(func):
     def inner_function(*args, **kwargs):
         try:
             return func(*args, **kwargs)
-        except pydantic.ValidationError as exc:
-            raise ValidationError("Not able to parse response from external service.") from exc
+        except httpx.ConnectError as exc:
+            raise ExternalServiceError("External service is not available.") from exc
```

Comment:
I'd Subclass a ExternalConnectionError from the ExternalServiceError and raise raise it here. You can still catch the more generic ExternalServiceError if you dont hava a way to handle the more specific ExternalConnectionError.

Reply:
Yeah, that's helpful! Will also do that for the validation error.